### PR TITLE
types: migrate test-infra to testify for time_test.go

### LINKED ISSUE #28010: migrate test-infra to testify for `time_test.go`

### PR DESCRIPTION
**Pull request by karuppiah7890**
### What problem does this PR solve?

Issue Number: close #28010 

Problem Summary:

### What is changed and how it works?

What's Changed:
- Migrated 45 test functions in `types/time_test.go` from `github.com/pingcap/check` to `github.com/stretchr/testify`
- Parallelized all the tests
- Removed usage of `github.com/pingcap/tidb/util/testleak` package code and the import

Tests

- [x] Unit test

### Release note 

```release-note
None
```

### Special Notes

I had to change variable names at times, as there were variable name clashes for `t` the variable generally used for golang test's `t *testing.T` vs `t` used in `time_test.go` for denoting `types.Time` or `types.Duration`. Sometimes I used `t1`, `t2` etc, sometimes `v`, sometimes `duration`. I noticed only later that `d` (short for duration) was fine too. And I used `tt` for a single record of tables tests, `tt` being short for `table test` or it could be considered just a random name

Do check the variable names and let me know if it's all okay. I also named some variables `time`, which can be confusing when the test functions use golang `time` standard library too. [UPDATE] I have fixed the variable naming of `time` here - https://github.com/karuppiah7890/tidb/commit/89470217f7ebd07402448fbbedc08b2cc7feda3e based on review comments

I also removed an extra parameter in table tests in `TestParseWithTimezone` test - `parseChecker Checker`, as it's value was always `isNil` and it came from `github.com/pingcap/check`. I could have used `expectError bool` like in some tests, and used value `false` for all records. But since error is **not** expected in all the records, I just removed the parameter. If in the future there are additional records which expect error, then we can use `expectError bool` with `true` value for records which expect error and `false` for others and use `if` and `else` with `expectError` and do assertion using `require.Error` or `require.NoError`. Also, I wanted to ask what's the significance of this portion of code in `TestParseWithTimezone` -

```go
if err != nil {
	continue
}
```

As the statements before it would fail the test if there is an error, so that code is not necessary I think. Let me know what you folks think

### CODE REVIEW

File: types/time_test.go [hunk 1]
```diff
@@ -50,25 +45,25 @@ func (s *testTimeSuite) TestTimeEncoding(c *C) {
...
-		c.Check(t.Type(), Equals, tt.Type, Commentf("%d type failed.", ith))
-		c.Check(t.Fsp(), Equals, tt.Fsp, Commentf("%d fsp failed.", ith))
-		c.Check(t.Year(), Equals, tt.Year, Commentf("%d year failed.", ith))
-		c.Check(t.Month(), Equals, tt.Month, Commentf("%d month failed.", ith))
-		c.Check(t.Day(), Equals, tt.Day, Commentf("%d day failed.", ith))
-		c.Check(t.Hour(), Equals, tt.Hour, Commentf("%d hour failed.", ith))
-		c.Check(t.Minute(), Equals, tt.Minute, Commentf("%d minute failed.", ith))
-		c.Check(t.Second(), Equals, tt.Second, Commentf("%d second failed.", ith))
-		c.Check(t.Microsecond(), Equals, tt.Microsecond, Commentf("%d microsecond failed.", ith))
+		time := types.NewTime(ct, tt.Type, tt.Fsp)
```
Comment (mmyj) [inline types/time_test.go:48]: It's better to avoid to use std pkg name as a var name.

Comment (karuppiah7890) [reply types/time_test.go:48]: Okay! Makes sense! I'll change it to `v` maybe?

Comment (karuppiah7890) [reply types/time_test.go:48]: I was thinking about this clash actually. Missed about mentioning it in the PR!

Comment (karuppiah7890) [reply types/time_test.go]: I have fixed it and changed variable names named `time` to `v`. https://github.com/karuppiah7890/tidb/commit/89470217f7ebd07402448fbbedc08b2cc7feda3e

Comment (karuppiah7890) [reply types/time_test.go]: Resolving this

**Review by mmyj — APPROVED** [review]
lgtm

**Review by tisonkun — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**